### PR TITLE
Add SITL target for px4vision

### PR DESCRIPTION
**Pull request by Jaeyoung-Lim**
This PR adds a gazebo SITL target for the Holybro Px4vision. More information can be found about the vehicle [here](https://docs.px4.io/master/en/complete_vehicles/px4_vision_kit.html)

This PR adds a sitl airframe and updates PX4/sitl_gazebo submodule that includes https://github.com/PX4/sitl_gazebo/pull/440

**Testing**
The vehicle shown flying below
px4vision

### CODE REVIEW

File: ROMFS/px4fmu_common/init.d-posix/10030_px4vision [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+#
+# @name PX4Vision SITL
+# Holybro px4vision. Gazebo Only.
+#
+# @type Quadrotor
+#
+
+sh /etc/init.d/rc.mc_defaults
+
```
Comment (TSC21) [inline ROMFS/px4fmu_common/init.d-posix/10030_px4vision]: This is being initialized as a standard quadcopter. Shouldn't this be capable of using VIO and avoidance?

**Comment by dagar** [comment]
Any reason we can't share the actual airframe 4016? https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/airframes/4016_holybro_px4vision

**Comment by TSC21** [comment]
> Any reason we can't share the actual airframe 4016? https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/airframes/4016_holybro_px4vision

That makes sense to me as well.

**Comment by Jaeyoung-Lim** [comment]
@dagar @TSC21 I still think we need a separate airframe config for sitl models since the mechanical properties / actuator configurations are all different. Wouldn't it be harder to maintain a single file that works on both?

IMHO, having the ekf2 masks and configurations (such as COM_OBS_AVOID) set properly would be already useful for people developing on the vehicle

**Comment by bys1123** [comment]
@Jaeyoung-Lim Then, Is that better if add "sitl" in file name, in case other developer edit wrong file.

**Comment by Jaeyoung-Lim** [comment]
@bys1123 It is already under a different directory. If you want to change the name, this would need to propagate also to all the models and make targets, which I find unnecessary

**Comment by bys1123** [comment]
Understood, thanks!

**Comment by TSC21** [comment]
> @dagar @TSC21 I still think we need a separate airframe config for sitl models since the mechanical properties / actuator configurations are all different. Wouldn't it be harder to maintain a single file that works on both?
> 
> IMHO, having the ekf2 masks and configurations (such as COM_OBS_AVOID) set properly would be already useful for people developing on the vehicle

@Jaeyoung-Lim the usual procedure is to try to match the configuration on both sides, and if it's not possible, it needs to be checked what parameters make sense or not and what parameters need an adjustment. My recommendation is that you copy-paste the values of one config file to the posix one and test it. If it works right, then you can keep it.

File: platforms/posix/cmake/sitl_target.cmake [hunk 2]
```diff
@@ -61,7 +61,7 @@ ExternalProject_Add(mavsdk_tests
 set(viewers none jmavsim gazebo)
 set(debuggers none ide gdb lldb ddd valgrind callgrind)
 set(models none shell
-	if750a iris iris_opt_flow iris_opt_flow_mockup iris_vision iris_rplidar iris_irlock iris_obs_avoid iris_rtps solo typhoon_h480
+	if750a iris iris_opt_flow iris_opt_flow_mockup iris_vision iris_rplidar iris_irlock iris_obs_avoid iris_rtps solo typhoon_h480 px4vision
```
Comment (TSC21) [inline platforms/posix/cmake/sitl_target.cmake]: Please make sure that we keep the alphabetical order here.

**Comment by dagar** [comment]
> @dagar @TSC21 I still think we need a separate airframe config for sitl models since the mechanical properties / actuator configurations are all different. Wouldn't it be harder to maintain a single file that works on both?

The goal would be to use that as a mechanism that forces the model to come into alignment with the real vehicle. If that's not realistic (or of interest) then maintaining both is fine.

**Comment by bys1123** [comment]
@Jaeyoung-Lim Any Update?

**Comment by bys1123** [comment]
@Jaeyoung-Lim Don't forget this, thanks~

File: ROMFS/px4fmu_common/init.d-posix/10030_px4vision (hunk 1 above)
Comment (Jaeyoung-Lim) [reply ROMFS/px4fmu_common/init.d-posix/10030_px4vision]: Done!

**Comment by Jaeyoung-Lim** [comment]
I tried using the airframe configs of the real vehicle and maybe override the necessary changes on the sitl airframe if necessary. This will make the diff more explicit and manageable. However, the airframe configs seems to work out of the box for the px4vision.

Would this be a acceptable way to add the configs? I am not sure if this adds complexity to the configuration.

**Comment by bys1123** [comment]
@Jaeyoung-Lim Thanks very much! looks good to me.

**Comment by bys1123** [comment]
@Jaeyoung-Lim @Jaeyoung-Lim Thanks very much! How about keep 
```
# param set COM_OBS_AVOID 1
# param set CP_DIST 6.0
```
but comment it, when user need it could easy to enable it?

**Comment by Jaeyoung-Lim** [comment]
@bys1123 The parameters are already enabled on the vehicle airframe configs here: https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/airframes/4016_holybro_px4vision#L29 , https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/airframes/4016_holybro_px4vision#L82

File: platforms/posix/cmake/sitl_target.cmake (hunk 2 above)
Comment (Jaeyoung-Lim) [reply platforms/posix/cmake/sitl_target.cmake]: Done!

**Comment by bys1123** [comment]
But not showing vision feature is ready to use on SITL side. 
Developers need to check model file again which in very deep directory. And check https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/airframes/4016_holybro_px4vision and PX4 Vision docs.
New users do not have the whole graph about the structures. But more tips in code could help them.

Just an user-friendly thinking, you could skip this.

**Comment by bys1123** [comment]
OK, Thanks!

**Comment by TSC21** [comment]
@Jaeyoung-Lim does it make sense to make use of this model for the Avoidance tests in CI? The iris with cameras is just and adapted model for the purpose and could make sense to use a model that is aimed for CV/obstacle avoidance instead.

**Comment by Jaeyoung-Lim** [comment]
@TSC21 I think that makes total sense. Would also get testing with real vehicles closer to the real vehicle

**Comment by TSC21** [comment]
> @TSC21 I think that makes total sense. Would also get testing with real vehicles closer to the real vehicle

Is that something you can explore in this PR as well?

**Comment by Jaeyoung-Lim** [comment]
@TSC21 Would it be okay if I address it in a separate PR? I won't be able to work on it for the next few days, so I would rather close this quickly.

**Comment by TSC21** [comment]
@Jaeyoung-Lim yes that's fine. I can't merge because it is stating that the branch has conflicts.

**Comment by Jaeyoung-Lim** [comment]
@TSC21 Hmm that is strange. I resolved it with https://github.com/PX4/Firmware/pull/14469/commits/557782fab042b549379092b48527fb81e58a243c

**Comment by TSC21** [comment]
> @TSC21 Hmm that is strange. I resolved it with [557782f](https://github.com/PX4/Firmware/commit/557782fab042b549379092b48527fb81e58a243c)

It must be Github doing its tricks again --'

**Comment by Jaeyoung-Lim** [comment]
@dagar Can we get this merged please?

**Comment by dagar** [comment]
> @dagar Can we get this merged please?

deja vu 😏

**Comment by Jaeyoung-Lim** [comment]
@dagar Haha thanks!

**Pull request merged**